packages/manager/apps/okms/src/common/hooks/useProductType.ts

```diff
@@ -0,0 +1,25 @@
+import { useLocation } from 'react-router-dom';
+
+export type ProductType = 'key-management-service' | 'secret-manager';
+
+const useProductType = (): ProductType | null => {
+  const location = useLocation();
+
+  // Split the pathname by '/' and get the first non-empty segment
+  const pathSegments = location.pathname
+    .split('/')
+    .filter((segment) => segment);
+  const productType = pathSegments[0];
+
+  // Ensure the productType is one of the allowed values
+  if (
+    productType !== 'key-management-service' &&
+    productType !== 'secret-manager'
+  ) {
+    return null;
+  }
+
+  return productType;
+};
+
+export default useProductType;
```

Comment:
Nitpick: I would have coded like this.

It is a bit more verbose but : 
- You don't write the type list twice
- You use a proper typegard for type narrowing

```suggestion
import { useLocation } from 'react-router-dom';

const productTypes = ['key-management-service', 'secret-manager'] as const;

export type ProductType = typeof productTypes[number];

const isProductType = (productType: string): productType is ProductType => {
  return productTypes.includes(productType as ProductType);
};

const useProductType = (): ProductType | null => {
  const location = useLocation();

  // Split the pathname by '/' and get the first non-empty segment
  const pathSegments = location.pathname
    .split('/')
    .filter((segment) => segment);
  const productType = pathSegments[0];

  // Ensure the productType is one of the allowed values
  if (!isProductType(productType)) {
    return null;
  }

  return productType;
};

export default useProductType;

```